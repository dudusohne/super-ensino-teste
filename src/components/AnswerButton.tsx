import { answerCharIndex } from '../pages/Home'

interface Props {
    isActive?: boolean;
    answer: string;
    onClickAnswer?: () => void;
}

export function AnswerButton(props: Props) {
    return (
        <button
            onClick={props.onClickAnswer}
            style={{
                display: 'flex',
                flexDirection: 'row',
                background: 'none',
                border: 'none',
            }}>
            <div style={{
                padding: '6px',
                backgroundColor: `${props.isActive ? 'blue' : 'gray'}`,
                borderRadius: '3px'
            }}>
                a
            </div>
            <span>{props.answer}</span>
        </button>
    )
}