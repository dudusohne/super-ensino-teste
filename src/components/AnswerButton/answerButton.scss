.answerButton {
  display: flex;
  flex-direction: row;
  background: none;
  border: none;

  margin-top: 5px;
}
