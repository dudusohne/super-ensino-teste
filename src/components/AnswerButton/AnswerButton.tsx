import React from 'react'

import { Answer } from '../../types';

import './answerButton.scss';

interface Props extends Answer {
    isActive?: boolean;
    onClickAnswer?: () => void;
}

const options = ['a', 'b', 'c', 'd']

export function AnswerButton({ isActive, answer, onClickAnswer, index }: Props) {
    return (
        <button
            className="answerButton"
            onClick={onClickAnswer}>
            <div className="answer-letter-wrapper" style={{ backgroundColor: `${isActive ? '#38d0fc' : '#cfcfcf'}`, }}>
                <p style={{ color: `${isActive ? '#07333f' : '#7a7a7a'}`, }}>
                    {options[index]}
                </p>
            </div>
            <span>{answer}</span>
        </button>
    )
}