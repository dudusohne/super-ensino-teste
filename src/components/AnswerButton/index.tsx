import React from 'react'

import './answerButton.scss';

interface Props {
    isActive?: boolean;
    answer: string;
    onClickAnswer?: () => void;
    index?: any;
}

const letters = ['a', 'b', 'c', 'd']

export function AnswerButton({isActive, answer, onClickAnswer, index}: Props) {
    return (
        <button
            className="answerButton"
            onClick={onClickAnswer}>
            <div className="answer-letter-wrapper" style={{ backgroundColor: `${isActive ? '#38d0fc' : '#cfcfcf'}`, }}>
                <p style={{ color: `${isActive ? '#07333f' : '#7a7a7a'}`, }}>
                    {letters[index]}
                </p>
            </div>
            <span>{answer}</span>
        </button>
    )
}