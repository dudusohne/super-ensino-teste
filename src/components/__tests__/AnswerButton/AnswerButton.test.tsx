import React from 'react'
import renderer from 'react-test-renderer'
import { AnswerButton } from '../../AnswerButton'

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}

test('create AnswerButton', () => {
  const component = renderer.create(
    <AnswerButton answer="resposta" index={1} />
  )
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()

  // re-rendering
  tree = toJson(component)
  expect(tree).toMatchSnapshot()
})