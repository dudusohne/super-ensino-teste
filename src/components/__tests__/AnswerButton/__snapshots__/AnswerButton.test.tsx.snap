// Vitest Snapshot v1

exports[`create AnswerButton 1`] = `
<button
  className="answerButton"
>
  <div
    className="answer-letter-wrapper"
    style={
      {
        "backgroundColor": "#cfcfcf",
      }
    }
  >
    <p
      style={
        {
          "color": "#7a7a7a",
        }
      }
    />
  </div>
  <span>
    resposta
  </span>
</button>
`;

exports[`create AnswerButton 2`] = `
<button
  className="answerButton"
>
  <div
    className="answer-letter-wrapper"
    style={
      {
        "backgroundColor": "#cfcfcf",
      }
    }
  >
    <p
      style={
        {
          "color": "#7a7a7a",
        }
      }
    />
  </div>
  <span>
    resposta
  </span>
</button>
`;
